dig.go

```diff
@@ -498,9 +519,23 @@ func (c *Container) provide(ctor interface{}, opts provideOptions) error {
 		}
 		c.isVerifiedAcyclic = true
 	}
-
 	c.nodes = append(c.nodes, n)
 
+	// Record introspection info for caller if Info option is specified
+	if opts.Info != nil {
+		opts.Info.ID = (int)(n.id)
+		opts.Info.Inputs = make([]string, 0)
+		opts.Info.Outputs = make([]string, 0)
```

Comment:
This can be shortened with `:=` inside if:
```suggestion
	if info := opts.Info; info != nil {
		info.ID = (int)(n.id)
		info.Inputs = make([]string, 0)
		info.Outputs = make([]string, 0)
```